jerry-core/parser/regexp/re-parser.cpp

```diff
@@ -27,150 +27,157 @@
 /**
  * Lookup a character in the input string.
  *
- * @return unicode codeunit
+ * @return true, if callback function holds true for lookup number of character ahead
+ *         false, otherwise
  */
-static ecma_char_t
+static bool
 re_lookup (lit_utf8_iterator_t iter, /**< input string iterator */
-           uint32_t lookup) /**< size of lookup */
+           uint32_t lookup, /**< size of lookup */
+           bool (*char_is_digit_function) (ecma_char_t)) /**< callback function */
```

Comment:
Am I understand correctly, that only hex digit is used for char_is_digit_function? I think we don't need specialization here then. Just a re_hex_peek() function